### PR TITLE
Fix some usage of deprecated `imp` module

### PR DESCRIPTION
**Pull request by asottile**
## Summary of changes

Partial resolution of #479 -- the other usage is a little more tricky and will require some more clever fixes

### Pull Request Checklist
- [x] Changes have tests
- [x] News fragment added in changelog.d. See [documentation](http://setuptools.readthedocs.io/en/latest/developer-guide.html#making-a-pull-request) for details

### CODE REVIEW

**Comment by asottile** [comment]
not sure why travis didn't report status -- it is complete and successful :man_shrugging:

File: setuptools/command/install_lib.py [hunk 1]
```diff
@@ -74,10 +74,10 @@ def _gen_exclusion_paths():
         yield '__init__.pyc'
         yield '__init__.pyo'
 
-        if not hasattr(imp, 'get_tag'):
+        if not hasattr(sys, 'implementation'):
             return
 
-        base = os.path.join('__pycache__', '__init__.' + imp.get_tag())
+        base = os.path.join('__pycache__', '__init__.' + sys.implementation.cache_tag)
```
Comment (mgedmin) [inline setuptools/command/install_lib.py]: The docs say `sys.implementation.cache_tag` may be `None` (in which case you're supposed to disable module caching).  Maybe check for that too and do an early return?

Comment (asottile) [reply setuptools/command/install_lib.py]: this isn't a regression, `imp.get_tag()` has identical semantics and is implemented as just `return sys.implementation.cache_tag`: https://github.com/python/cpython/blob/9b5ce62cac27fec9dea473865d79c2c654312957/Lib/imp.py#L67-L69

**Comment by pganssle** [comment]
@asottile I don't have any objections to this, but when running the test suite against master with `DeprecationWarning` set to raise errors on `master`, I'm not getting any failures, but `coverage.py` is saying all these lines are covered. Any idea what's going on there?

**Comment by asottile** [comment]
> @asottile I don't have any objections to this, but when running the test suite against master with `DeprecationWarning` set to raise errors on `master`, I'm not getting any failures, but `coverage.py` is saying all these lines are covered. Any idea what's going on there?

depending on the version of `pytest` you're using (I've only fixed pytest in 5.x) `imp` will get imported as a side-effect of pytest's initialization (and so imports in user-space code won't trigger that warning) -- this is probably what you're seeing if I had to guess?

**Comment by pganssle** [comment]
Hm.. That seems like a huge problem with `pytest`, that's super annoying.

**Comment by asottile** [comment]
hehe yeah that's why I fixed it 🙃

**Comment by asottile** [comment]
@pganssle @jaraco any updates on this? I think this is ready for merge

**Pull request merged**